iree/hal/vulkan/serializing_command_queue.cc

```diff
@@ -203,13 +203,33 @@ StatusOr<bool> SerializingCommandQueue::ProcessDeferredSubmissions() {
   // A list of submissions that still needs to be deferred.
   IntrusiveList<std::unique_ptr<FencedSubmission>> remaining_submissions;
 
+  // We need to return all remaining submissions back to the queue to avoid
+  // dropping work.
+  auto submission_cleanup = MakeCleanup([this, &remaining_submissions]() {
+// Disable thread-safety-analysis as it doesn't understand this lambda.
```

Comment:
For the record, I'm not totally confident in my interpretation of the analysis here, and this pattern isn't listed as a ["known-limitation"](https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#known-limitations).